src/Protocol/DFRN.php

```diff
@@ -956,10 +956,9 @@ private static function entry(DOMDocument $doc, $type, array $item, array $owner
 		$entry->appendChild($dfrnowner);
 
 		if ($item['gravity'] != GRAVITY_PARENT) {
-			$parent_item = (($item['thr-parent']) ? $item['thr-parent'] : $item['parent-uri']);
-			$parent = Item::selectFirst(['guid', 'plink'], ['uri' => $parent_item, 'uid' => $item['uid']]);
+			$parent = Item::selectFirst(['guid', 'plink'], ['uri' => $item['thr-parent'], 'uid' => $item['uid']]);
 			if (DBA::isResult($parent)) {
-				$attributes = ["ref" => $parent_item, "type" => "text/html",
+				$attributes = ["ref" => $item['thr-parent'], "type" => "text/html",
```

Comment:
Same like in the feeds: We should create an update function to fill empty `thr-parent` fields.